### DISCUSSION TITLE
Testando o Handler

### DISCUSSION THREAD

**Discussion by maxaraujo** [balta-io/7182#3] [Perguntas e Respostas] [answered]
Estava implementando o testes do curso e fiquei com uma dúvida sobre o último teste feito. 
Mesmo passando vários valores nulos para o teste Dado_um_comando_valido_o_pedido_deve_ser_gerado, ele ainda está passando. Isso deveria acontecer?

        [TestMethod]
        [TestCategory("Handlers")]
        public void Dado_um_comando_valido_o_pedido_deve_ser_gerado()
        {
            var command = new CreateOrderCommand();
            command.Customer = null;
            command.ZipCode = "13411080";
            command.PromoCode = "12345678";

            var handler = new OrderHandler(
                null,
                null,
                null,
                null,
                null);

            handler.Handle(command);
            Assert.AreEqual(handler.Valid, true);
        }

**Comment by andrebaltieri** [comment, marked as answer]
Boa tarde @maxaraujo poderia postar o código do seu handler?

**Reply by maxaraujo** [reply]
Boa noite, @andrebaltieri! Eu também tinha achado que o problema estava no meu handler, então baixei e testei o código pronto e aconteceu a mesma coisa. De qualquer forma, segue meu código: 

using System.Linq;
using Flunt.Notifications;
using Store.Domain.Commands;
using Store.Domain.Commands.Interfaces;
using Store.Domain.Entities;
using Store.Domain.Handlers.Interfaces;
using Store.Domain.Repositories;
using Store.Domain.Utils;

namespace Store.Domain.Handlers
{
    public class OrderHandler :
        Notifiable,
        IHandler<CreateOrderCommand>
    {
        private readonly ICustomerRepository _customerRepository;
        private readonly IDeliveryFeeRepository _deliveryFeeRepository;
        private readonly IDiscountRepository _discountRepository;
        private readonly IProductRepository _productRepository;
        private readonly IOrderRepository _orderRepository;

        public OrderHandler(
            ICustomerRepository customerRepository,
            IDeliveryFeeRepository deliveryFeeRepository,
            IDiscountRepository discountRepository,
            IOrderRepository orderRepository)
        {
            _customerRepository = customerRepository;
            _deliveryFeeRepository = deliveryFeeRepository;
            _discountRepository = discountRepository;
            _orderRepository = orderRepository;
        }
        public ICommandResult Handle(CreateOrderCommand command)
        {
            command.Validate();
            if(command.Invalid)
                return new GenericCommandResult(false, "Pedido inválido", command.Notifications);

            var customer = _customerRepository.Get(command.Customer);

            var deliveryFee = _deliveryFeeRepository.Get(command.ZipCode);

            var discount = _discountRepository.Get(command.PromoCode);

            var products = _productRepository.Get(ExtractGuids.Extract(command.Items)).ToList();

            var order = new Order(customer, deliveryFee, discount);

            foreach(var item in command.Items)
            {
                var product = products.Where(x => x.Id == item.Product).FirstOrDefault();
                order.AddItem(product, item.Quantity);
            }

            AddNotifications(customer.Notifications);
            AddNotifications(order.Notifications);

            if(Invalid)
                return new GenericCommandResult(false, "Falha ao gerar o pedido", Notifications);

            _orderRepository.Save(order);
            return new GenericCommandResult(true, $"Pedido {order.Number} gerada com sucesso", order);
        }
    }
}

**Reply by andrebaltieri** [reply]
@maxaraujo show, obrigado por compartilhar! Acho que não realizei as validações então... mas deveria falhar sim... no caso só colocar os devidos `if`s!

```csharp
if(model == null)
```
